### ISSUE TITLE
Confusing error msg if path in "packages" doesn't exist

### ISSUE THREAD

**Issue by geckon** [python-poetry/poetry#2139]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request

Could poetry please handle typos/errors in `tool.poetry.packages` better?

If you make a typo or another error in specifying packages to include into your package and the modules specified in pyproject.toml don't exist, you'll get a very user-unfriendly message:
```
$ poetry build
Building dummy_package (1.49.1)

[IndexError]
list index out of range
```

Let's say you have a package called dummy_package and in it you have python modules dummy1 and dummy2 you want to include into the package. Then you'd like to put this into your pyproject.toml:
```
packages = [
    { include = "dummy1" },
    { include = "dummy2" },    
]
```
If instead you make a mistake and have e.g. this there:
```
packages = [
    { include = "dumny1" },
    { include = "dumny2" },    
]
```
You ge tthe above mentioned error which is very confusing and not helpful at all.

The error happens because you mention a module that doesn't exist in your project directory. The actual error happens in method `check_elements()` (specifically on line 37 in current master) in `poetry/masonry/utils/package_include.py` but I think it might be detectable in `poetry/masonry/utils/include.py` where it tries to expand the path. Maybe there it could issue a warning or something?

The issue was mentioned e.g. in  #1247 but that issue is closed and I'd like this to get a little more attention as I spent a few hours debugging this.

Thank you!

**Comment by autoferrit** [comment]
I have the same issue. for me I believe it is because of this portion of my `pyproject.toml`
```toml
packages = [
    { include = "bohemian", from = "src"}
]
```
I am mostly doing this, because my project is inside a `src` directory. This happens in CI when I add the `pyproject.toml` and `poetry.lock` into a container and run install, when my apps code isnt there yet.
But it work when I install with 
```bash
poetry install --no-root
```
and then don't install or run `poetry build` until I add my code into the repo.

**Comment by aaugustin** [comment]
I have been hitting this error as well.

I considered moving to a src layout, then changed my mind, and eventually ran poetry with `from = "src"` in my `pyproject.toml` while my package wasn't in a `src` directory.

This is clearly a user error on my side but the error message is less than helpful and the traceback with `-v` isn't any better.

**Comment by PierreAndreNoel** [comment]
Got the same meaningless error, but should be fixed by #2265

**Comment by mkniewallner** [comment]
As already mentioned, this was fixed by https://github.com/python-poetry/poetry/pull/2265.
